composeml/label_times/object.py

```diff
@@ -17,65 +17,84 @@ def __init__(
+        if not self.empty: self._check_label_times()
+
+    def _assert_single_target(self):
+        info = 'must first select an individual target'
+        assert self._is_single_target, info
 
-    def _check_label_name(self):
+    def _check_label_times(self):
         """Checks whether the target exists in the data frame."""
-        if self.label_name is None:
-            self.label_name = self._infer_label_name()
+        if not self.target_columns:
+            self.target_columns = self._infer_target_columns()
+
+        missing = pd.Index(self.target_columns).difference(self.columns)
+        info = 'target variable(s) not found: %s'
+        assert missing.empty, info % missing.tolist()
+
+        if not isinstance(self.target_types, pd.Series):
+            self.target_types = pd.Series(self.target_types)
 
-        info = 'target variable not found: %s' % self.label_name
-        assert self.label_name in self.columns, info
+        if self.target_types.empty:
+            self.target_types = self._infer_target_types()
 
-    def _infer_label_name(self):
+    def _infer_target_columns(self):
         """Infers the target name from the data frame.
 
         Returns:
             value (str): Inferred target name.
         """
         not_targets = [self.target_entity, 'time']
         target_names = self.columns.difference(not_targets)
-        value = target_names.tolist()[0]
+        value = target_names.tolist()
         return value
 
-    def _check_label_type(self):
-        """Checks whether the target type is continuous or discrete."""
-        if self.label_type is None:
-            self.label_type = self._infer_label_type()
-
-        error = 'label type must be "continuous" or "discrete"'
-        assert self.label_type in ['continuous', 'discrete'], error
-
-    def _infer_label_type(self):
-        """Infers the target type from the data type.
+    @property
+    def _is_single_target(self):
+        return len(self.target_columns) == 1
 
-        Returns:
-            value (str): Inferred label type. Either "continuous" or "discrete".
-        """
-        dtype = self[self.label_name].dtype
+    def _get_target_type(self, dtype):
         is_discrete = pd.api.types.is_bool_dtype(dtype)
         is_discrete |= pd.api.types.is_categorical_dtype(dtype)
         is_discrete |= pd.api.types.is_object_dtype(dtype)
         value = 'discrete' if is_discrete else 'continuous'
         return value
 
+    def _infer_target_types(self):
+        """Infers the target type from the data type.
+
+        Returns:
+            types (Series): Inferred label type. Either "continuous" or "discrete".
+        """
+        dtypes = self.dtypes[self.target_columns]
+        types = dtypes.apply(self._get_target_type)
+        return types
+
+    def select(self, target):
+        assert not self._is_single_target, 'only one target exists'
+        assert target in self.target_columns, 'target "%s" not found' % target
+
```

Comment:
When a user calls `select()` I think we are expecting a string to be passed in. Right now it is possible to pass in any type of object. Passing something like an `int` or a `list` will cause the assertion to fail: `assert target in self.target_columns, 'target "%s" not found' % target`, but it might be better to specifically check that the passed target is a `str` and raise a `TypeError` if it is not. This may be kind of an edge case, but a user could have a column named 123 and if they pass in the integer 123 they will get an error saying `target "123" not found`  because the assertion is checking if the integer 123 is in the string column names, and that could be confusing.